src/cli_plugin/install/__tests__/download.js

```diff
@@ -0,0 +1,226 @@
+import expect from 'expect.js';
+import sinon from 'sinon';
+import nock from 'nock';
+import glob from 'glob-all';
+import rimraf from 'rimraf';
+import mkdirp from 'mkdirp';
+import Logger from '../../lib/logger';
+import { download, _downloadSingle } from '../download';
+import { join } from 'path';
+
+describe('kibana cli', function () {
+
+  describe('plugin downloader', function () {
+    const testWorkingPath = join(__dirname, '.test.data');
+    const tempArchiveFilePath = join(testWorkingPath, 'archive.part');
+
+    const settings = {
+      urls: [],
+      workingPath: testWorkingPath,
+      tempArchiveFile: tempArchiveFilePath,
+      timeout: 0
+    };
+    const logger = new Logger(settings);
+
+    function expectWorkingPathEmpty() {
+      const files = glob.sync('**/*', { cwd: testWorkingPath });
+      expect(files).to.eql([]);
+    }
+
+    function expectWorkingPathNotEmpty() {
+      const files = glob.sync('**/*', { cwd: testWorkingPath });
+      const expected = [
+        'archive.part'
+      ];
+
+      expect(files.sort()).to.eql(expected.sort());
+    }
+
+    function shouldReject() {
+      throw new Error('expected the promise to reject');
+    }
+
+    beforeEach(function () {
+      sinon.stub(logger, 'log');
+      sinon.stub(logger, 'error');
+      rimraf.sync(testWorkingPath);
+      mkdirp.sync(testWorkingPath);
+    });
+
+    afterEach(function () {
+      logger.log.restore();
+      logger.error.restore();
+      rimraf.sync(testWorkingPath);
+    });
+
+    describe('_downloadSingle', function () {
+
+      beforeEach(function () {
+      });
+
+      describe('http downloader', function () {
+
+        it('should throw an ENOTFOUND error for a http ulr that returns 404', function () {
+          const couchdb = nock('http://www.files.com')
```

Comment:
We should be using `example.com` for anything like this.  In there rare event that we introduce a bug that makes our tests actually call out to whatever url we specified, we don't want to be spamming a real domain.